apps/api/v2/src/modules/oauth/guard/oauth-client/oauth-client.guard.ts

```diff
@@ -1,29 +1,40 @@
 import { OAuthClientRepository } from "@/modules/oauth/oauth-client.repository";
-import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
+import { CanActivate, ExecutionContext, Injectable, UnauthorizedException } from "@nestjs/common";
+import { PlatformOAuthClient } from "@prisma/client";
 
-import { X_CAL_CLIENT_ID, X_CAL_SECRET_KEY } from "@calcom/platform-constants";
+import { X_CAL_SECRET_KEY } from "@calcom/platform-constants";
+
+export type AuthenticatedRequest = Request & {
+  oAuthClient: PlatformOAuthClient;
+};
 
 @Injectable()
 export class OAuthClientGuard implements CanActivate {
   constructor(private readonly oauthRepository: OAuthClientRepository) {}
 
-  canActivate(context: ExecutionContext): Promise<boolean> {
+  async canActivate(context: ExecutionContext): Promise<boolean> {
     const request = context.switchToHttp().getRequest();
-    const { headers } = request;
+    const { headers, params } = request;
 
-    const oauthClientId = headers[X_CAL_CLIENT_ID];
+    const oauthClientId = params.clientId;
     const oauthClientSecret = headers[X_CAL_SECRET_KEY];
 
-    return this.validateOauthClient(oauthClientId, oauthClientSecret);
-  }
+    if (!oauthClientId) {
+      throw new UnauthorizedException("Missing client ID");
+    }
 
-  private async validateOauthClient(oauthClientId: string, oauthClientSecret: string): Promise<boolean> {
-    const oauthClient = await this.oauthRepository.getOAuthClient(oauthClientId);
+    if (!oauthClientSecret) {
+      throw new UnauthorizedException("Missing client secret");
+    }
+
+    const client = await this.oauthRepository.getOAuthClient(oauthClientId);
 
-    if (!oauthClient || oauthClient.secret !== oauthClientSecret) {
-      return false;
+    if (!client || client.secret !== oauthClientSecret) {
+      throw new UnauthorizedException();
     }
 
+    request.oAuthClient = client;
```

Comment:
we dont need to type the request or create a decorator for this ?

Reply:
Removed in [remove oauth client from request](https://github.com/calcom/cal.com/pull/12853/commits/5817d57f01f5b36d5dae951aad6d5f59f1999e4d)